lib/datadog/tracing/trace_operation.rb

```diff
@@ -417,6 +429,26 @@ def set_root_span!(span)
         @root_span = span
       end
 
+      # Flushes all spans from this trace
+      def flush_all_spans!
+        # Copy out completed spans
+        spans = @spans.dup
+        @spans = []
+
+        # Use them to build a trace
+        build_trace(spans, !finished?)
+      end
+
+      # Flush single sampled span only, as the trace as a whole was dropped by trace-level sampling
+      def flush_single_sampled_spans_only!
```

Comment:
This logic seems very duplicative of the above/original logic. Should we be be adding an option to `flush!` that allows both of these paths to be driven based on optional input?